packages/inferno-mobx/src/observerPatch.ts

```diff
@@ -13,10 +13,19 @@ function makeObserverRender<R extends Render>(update: () => void, render: R, nam
   const track = reactor.track.bind(reactor);
   const observer = function (this, ...parameters: Parameters<typeof render>) {
     let rendered: RenderReturn;
+    let caught;
     track(() => {
-      rendered = render.apply(this, parameters);
+      try {
```

Comment:
can we have a test case for this please? This will be expensive :/

Reply:
@Havunen 

I will add those tests for both `observer` and `observerPatch`.

> This will be expensive :/

The `observer` function does this and also sends the exception to an error event handler. When I first looked through it I had thought it only sent them to the event emitter, which MobX already does something similar if we do not catch it. This is also how the current version of mobx-react deals with exceptions from `render`.

The reason for doing this is to make the observer component act the same as if it had not been made into an observer. If it is fine for it to be different this can be left out. It could also be made optional and allow the behavior to be selected when calling `observerPatch`.

On the other hand, having render throw exceptions at all is questionable. If the error occurs in response to the MobX reaction of an observer (and force update does not become async) the error will still end up going to the MobX global handler once is unwinds past the forceUpdate call. And any time a state change becomes async, it will result in an unhandled exception in a promise unless Inferno catches it somewhere.

Reply:
ok, LGTM. thanks for another PR! :)